component/src/main/java/org/wso2/extension/siddhi/io/file/util/FileSourceConfiguration.java

```diff
@@ -0,0 +1,226 @@
+/*
+ * Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.extension.siddhi.io.file.util;
+
+import org.wso2.carbon.messaging.CarbonMessageProcessor;
+import org.wso2.carbon.messaging.ServerConnector;
+import org.wso2.carbon.transport.file.connector.server.FileServerConnector;
+
+import java.util.concurrent.Executor;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+/**
+ * Class for keep the configurations of a file source instance.
+ * */
+public class FileSourceConfiguration {
+
+    private String actionAfterProcess;
+    private String moveAfterProcessUri;
+    private boolean isTailingEnabled;
+    private String dirURI;
+    private String fileURI;
+    private String mode;
+    private String beginRegex = null;
+    private String endRegex = null;
+    private String filePointer = "0";
```

Comment:
Why does this `filePointer` has to be a string?

Reply:
Carbon file transport requires a <string, string> map for server connector properties. the file pointer is also passed as such property.

https://github.com/wso2/transport-file/blob/master/components/org.wso2.transport.file/src/main/java/org/wso2/transport/file/connector/server/FileServerConnectorProvider.java#L46